mitmproxy/controller.py

```diff
@@ -147,23 +167,63 @@ def __call__(self, msg=False):
 NO_REPLY = object()
 
 
-class Reply(object):
+def handler(f):
+    @functools.wraps(f)
+    def wrapper(*args, **kwargs):
+        if len(args) == 1:
```

Comment:
What funky magic are we doing here?
(From reading the diff sequentially until here I suspect that this deals with inheritance, but we should add a comment whatsoever)